Cubical/Foundations/HLevels.agda

```diff
@@ -423,3 +423,56 @@ isOfHLevel→isOfHLevelDep (suc (suc n)) {A = A} {B} h {a0} {a1} b0 b1 =
   helper a1 p b1 = J
                      (λ a1 p → ∀ b1 → isOfHLevel (suc n) (PathP (λ i → B (p i)) b0 b1))
                      (λ _ → h _ _ _) p b1
+
+isContrDep→isPropDep : isOfHLevelDep 0 B → isOfHLevelDep 1 B
+isContrDep→isPropDep {B = B} Bctr {a0 = a0} b0 b1 p i
+  = comp (λ k → B (p (i ∧ k))) (λ k → λ where
+        (i = i0) → Bctr .snd b0 refl k
+        (i = i1) → Bctr .snd b1 p k)
+      (c0 .fst)
+  where
+  c0 = Bctr {a0}
+
+isPropDep→isSetDep : isOfHLevelDep 1 B → isOfHLevelDep 2 B
+isPropDep→isSetDep {B = B} Bprp b0 b1 b2 b3 p i j
+  = comp (λ k → B (p (i ∧ k) (j ∧ k))) (λ k → λ where
+        (j = i0) → Bprp b0 b0 refl k
+        (i = i0) → Bprp b0 (b2 j) (λ k → p i0 (j ∧ k)) k
+        (i = i1) → Bprp b0 (b3 j) (λ k → p k (j ∧ k)) k
+        (j = i1) → Bprp b0 b1 (λ k → p (i ∧ k) (j ∧ k)) k)
+      b0
+
+isOfHLevelDepSuc : (n : HLevel) → isOfHLevelDep n B → isOfHLevelDep (suc n) B
+isOfHLevelDepSuc 0 = isContrDep→isPropDep
+isOfHLevelDepSuc 1 = isPropDep→isSetDep
+isOfHLevelDepSuc (suc (suc n)) Blvl b0 b1 = isOfHLevelDepSuc (suc n) (Blvl b0 b1)
+
+isPropDep→isSetDep'
+  : isOfHLevelDep 1 B
+  → {p : w ≡ x} {q : y ≡ z} {r : w ≡ y} {s : x ≡ z}
+  → {tw : B w} {tx : B x} {ty : B y} {tz : B z}
+  → (sq : Square p q r s)
+  → (tp : PathP (λ i → B (p i)) tw tx)
+  → (tq : PathP (λ i → B (q i)) ty tz)
+  → (tr : PathP (λ i → B (r i)) tw ty)
+  → (ts : PathP (λ i → B (s i)) tx tz)
+  → SquareP (λ i j → B (sq i j)) tp tq tr ts
+isPropDep→isSetDep' {B = B} Bprp {p} {q} {r} {s} {tw} sq tp tq tr ts i j
+  = comp (λ k → B (sq (i ∧ k) (j ∧ k))) (λ k → λ where
+        (i = i0) → Bprp tw (tp j) (λ k → p (k ∧ j)) k
+        (i = i1) → Bprp tw (tq j) (λ k → sq (i ∧ k) (j ∧ k)) k
+        (j = i0) → Bprp tw (tr i) (λ k → r (k ∧ i)) k
+        (j = i1) → Bprp tw (ts i) (λ k → sq (k ∧ i) (j ∧ k)) k)
+      tw
+
+isOfHLevelΣ' : ∀ n → isOfHLevel n A → isOfHLevelDep n B → isOfHLevel n (Σ A B)
+isOfHLevelΣ' 0 Actr Bctr .fst = (Actr .fst , Bctr .fst)
+isOfHLevelΣ' 0 Actr Bctr .snd (x , y) i
+  = Actr .snd x i , Bctr .snd y (Actr .snd x) i
+isOfHLevelΣ' 1 Alvl Blvl (w , y) (x , z) i .fst = Alvl w x i
+isOfHLevelΣ' 1 Alvl Blvl (w , y) (x , z) i .snd = Blvl y z (Alvl w x) i
+isOfHLevelΣ' {A = A} {B = B} (suc (suc n)) Alvl Blvl (w , y) (x , z)
```

Comment:
Can you prove this using `isOfHLevelRetract` instead? See the following for inspiration:

https://github.com/agda/cubical/pull/421/files#diff-4b84591f239c56566629cde01652543eR306

When working on that PR @aljungstrom noticed that using `isOfHLevelRetract` seem to produce faster code further down the line